edgelet/support-bundle/src/shell_util.rs

```diff
@@ -0,0 +1,251 @@
+    verbose: bool,
+) -> Result<(), Error>
+where
+    W: Write + Seek,
+{
+    print_verbose(
+        &format!("Running docker network inspect for {}", network_name),
+        verbose,
+    );
+    let mut inspect = Command::new("docker");
+
+    inspect.args(&["network", "inspect", &network_name, "-v"]);
+    let inspect = inspect.output().await;
+
+    let (file_name, output) = if let Ok(result) = inspect {
+        if result.status.success() {
+            (format!("network/{}.json", network_name), result.stdout)
+        } else {
+            (format!("network/{}_err.json", network_name), result.stderr)
+        }
+    } else {
+        let err_message = inspect.err().unwrap().to_string();
+        println!(
+            "Could not reach docker. Including error in bundle.\nError message: {}",
+            err_message
+        );
+        (
+            format!("network/{}_err_docker.txt", network_name),
+            err_message.as_bytes().to_vec(),
+        )
+    };
+
+    zip_writer
+        .start_file(file_name, *file_options)
+        .map_err(|err| Error::from(err.context(ErrorKind::SupportBundle)))?;
+
+    zip_writer
+        .write_all(&output)
+        .map_err(|err| Error::from(err.context(ErrorKind::SupportBundle)))?;
+
+    print_verbose(
+        &format!("Got docker network inspect for {}", network_name),
+        verbose,
+    );
+    Ok(())
+}
+
+pub async fn write_system_log<W>(
+    name: &str,
+    unit: &str,
+    log_options: &LogOptions,
+    zip_writer: &mut ZipWriter<W>,
+    file_options: &FileOptions,
+    verbose: bool,
+) -> Result<(), Error>
+where
+    W: Write + Seek,
+{
+    print_verbose(
+        format!("Getting system logs for {}", name).as_str(),
+        verbose,
+    );
+    let since_time: DateTime<Utc> = DateTime::from_utc(
+        NaiveDateTime::from_timestamp(log_options.since().into(), 0),
+        Utc,
+    );
+    let until_time: Option<DateTime<Utc>> = log_options
+        .until()
+        .map(|until| DateTime::from_utc(NaiveDateTime::from_timestamp(until.into(), 0), Utc));
+
+    #[cfg(unix)]
+    let command = {
+        let mut command = Command::new("journalctl");
+        command
+            .arg("-a")
+            .args(&["-u", unit])
+            .args(&["-S", &since_time.format("%F %T").to_string()])
+            .arg("--no-pager");
+        if let Some(until) = until_time {
+            command.args(&["-U", &until.format("%F %T").to_string()]);
+        }
+
+        command.output().await
+    };
+
+    let (file_name, output) = if let Ok(result) = command {
+        if result.status.success() {
+            (format!("logs/{}.txt", name), result.stdout)
+        } else {
+            (format!("logs/{}_err.txt", name), result.stderr)
+        }
+    } else {
+        let err_message = command.err().unwrap().to_string();
+        println!(
+            "Could not find system logs for {}. Including error in bundle.\nError message: {}",
+            name, err_message
```

Comment:
Suggestion: Feel free to ignore, but might be nice for these prints that are not success cases if we could use escape codes and print out as red text or something?